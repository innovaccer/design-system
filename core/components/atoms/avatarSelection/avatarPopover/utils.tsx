import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>,
  listRef?: any,
  withSearch?: boolean,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'Enter':
      handleEnterKey(focusedOption);
      setHighlightLastItem?.(false);
      setHighlightFirstItem?.(false);
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }
};

const handleEnterKey = (focusedOption: Element | undefined) => {
  (focusedOption as HTMLElement)?.click();
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  const listItems = listRef?.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  if (index === -1) {
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else if (
    (withSearch && index === 0 && direction === 'up') ||
    (withSearch && index === listItems.length - 1 && direction === 'down')
  ) {
    const searchInput = listRef.current.querySelector('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput.focus();
    setFocusedOption && setFocusedOption(searchInput);
  } else {
    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;

    const targetOption = listItems[index];

    (targetOption as HTMLElement).focus();
    setFocusedOption && setFocusedOption(targetOption);
    targetOption.scrollIntoView({ block: 'center' });
  }
};

export const handleInputKeyDown = (
  event: React.KeyboardEvent,
  listRef: any,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      targetOption = listItems[listItems.length - 1];
      break;
    case 'ArrowDown':
      event.preventDefault();
      targetOption = listItems[0];
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }

  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
