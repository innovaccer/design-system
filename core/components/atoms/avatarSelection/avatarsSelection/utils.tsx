import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>
) => {
  switch (event.key) {
    case 'Enter':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowDown':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowUp':
      setOpenPopover?.(true);
      setHighlightLastItem?.(true);
      break;
    default:
      break;
  }
};

export const focusListItem = (
  position: string,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  if (withSearch && position === 'down') {
    const searchInput = listRef.current?.querySelectorAll('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput && searchInput[0]?.focus();
    setFocusedOption?.(searchInput);
    return;
  }

  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  if (position === 'down') {
    targetOption = listItems?.[0];
  } else {
    targetOption = listItems?.[listItems.length - 1];
  }
  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
