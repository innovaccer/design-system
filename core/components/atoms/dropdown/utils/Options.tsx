export let dropdownOptions: any[] = [];

for (let i = 1; i <= 50; i++) {
  dropdownOptions.push({
    label: `Option${i}`,
    value: `Option${i}`,
  });
}

export const selectedStoryOptions: any[] = [];

for (let i = 1; i <= 10; i++) {
  selectedStoryOptions.push({
    label: `Option${i}`,
    value: `Option${i}`,
    selected: i === 2,
  });
}

export const multiSelectedStoryOptions: any[] = [];

for (let i = 1; i <= 10; i++) {
  multiSelectedStoryOptions.push({
    label: `Option${i}`,
    value: `Option${i}`,
    selected: i === 2 || i === 7,
  });
}

export const storyOptions = dropdownOptions.slice(0, 10);

export const storySections = { 0: 'subheading', 5: 'subheading' };

export const subInfoItems = [
  {
    label: 'Option1',
    value: 'Option1',
    subInfo: 'subInfo'
  },
  {
    label: 'Option2',
    value: 'Option2',
    subInfo: 'subInfo'
  },
  {
    label: 'Option3',
    value: 'Option3',
    subInfo: 'subInfo',
  },
];

export const iconItems = [
  {
    label: 'Option1',
    value: 'Option1',
    icon: 'events'
  },
  {
    label: 'Option2',
    value: 'Option2',
    icon: 'events',
  },
  {
    label: 'Option3',
    value: 'Option3',
    icon: 'events',
  },
];
