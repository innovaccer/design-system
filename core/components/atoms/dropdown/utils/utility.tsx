const getSearchedOptions = (options: any, searchTerm: string) => {
  const result = options.filter((option: any) => option.label.toLowerCase().includes(searchTerm.toLowerCase()));
  return result;
};

export const getOptions = (offset: number, limit: number, searchTerm: string, options: any) => {
  const searchedOptions = searchTerm ? getSearchedOptions(options, searchTerm) : options;
  return new Promise(resolve => {
    resolve({
      offset,
      slicedOptions: searchedOptions.slice(offset, offset + limit),
      options: searchedOptions,
    });
  });
};

export const getValuesFromSelectedObj = (selectedArray: any[] = []) => {
  const selectedValues: any[] = [];

  selectedArray.forEach((selectedObj: any) => {
    const { value } = selectedObj;
    selectedValues.push(value);
  });

  return selectedValues;
};

export const getIndexesFromSelectedObj = (selectedArray: any[] = []) => {
  return (Array.from(Array(selectedArray.length).keys()));
};

export const getValuesFromInd = (selectedArray: number[], options: any[]) => {
  const result: any[] = [];
  selectedArray.forEach(ind => {
    result.push(options[ind].value);
  });

  return result;
};

export const getLabelsFromInd = (selectedArray: number[], options: any[]) => {
  const result: string[] = [];
  selectedArray.forEach(ind => {
    result.push(options[ind].label);
  });

  return result;
};
