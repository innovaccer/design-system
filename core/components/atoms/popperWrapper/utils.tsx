// Helper function to convert CSS class names to style objects
export const createStyleFromClass = (className: string) => {
  if (!className) return {};

  // Create a temporary element to get computed styles
  const tempElement = document.createElement('div');
  tempElement.className = className;
  document.body.appendChild(tempElement);

  const computedStyle = window.getComputedStyle(tempElement);
  const style: React.CSSProperties = {};

  // Extract relevant animation properties
  const animation = computedStyle.animation;
  const animationDuration = computedStyle.animationDuration;
  const animationTimingFunction = computedStyle.animationTimingFunction;
  const animationFillMode = computedStyle.animationFillMode;
  const opacity = computedStyle.opacity;
  const transform = computedStyle.transform;

  if (animation && animation !== 'none') {
    style.animation = animation;
  }
  if (animationDuration && animationDuration !== '0s') {
    style.animationDuration = animationDuration;
  }
  if (animationTimingFunction && animationTimingFunction !== 'ease') {
    style.animationTimingFunction = animationTimingFunction;
  }
  if (animationFillMode && animationFillMode !== 'none') {
    style.animationFillMode = animationFillMode;
  }
  if (opacity && opacity !== '1') {
    style.opacity = opacity;
  }
  if (transform && transform !== 'none') {
    style.transform = transform;
  }

  document.body.removeChild(tempElement);
  return style;
};
