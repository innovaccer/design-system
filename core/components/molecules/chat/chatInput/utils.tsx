import { MessageType } from './ChatInput';

export const positionMentionPopup = (
  textareaRef: React.RefObject<HTMLDivElement>,
  setMentionPosition: React.Dispatch<React.SetStateAction<{ top: number; left: number }>>
) => {
  if (!textareaRef.current) return;

  const selection = window.getSelection();
  if (!selection || !selection.rangeCount) return;

  const range = selection.getRangeAt(0);
  const rect = range.getBoundingClientRect();
  const editorRect = textareaRef.current.getBoundingClientRect();

  // Calculate position based on the cursor position
  let top = rect.bottom + window.scrollY;
  let left = rect.left + window.scrollX;

  // Adjust position if it would go off-screen
  if (top + 150 > window.innerHeight) {
    top = rect.top + window.scrollY - 150;
  }
  if (left + 200 > window.innerWidth) {
    left = window.innerWidth - 200;
  }

  // Adjust position relative to the editor
  top -= editorRect.top + window.scrollY - 12;
  left -= editorRect.left + window.scrollX - 8;

  setMentionPosition({ top, left });
};

export const extractMessageData = (textareaRef: React.RefObject<HTMLDivElement>): MessageType[] => {
  const elements = textareaRef.current?.childNodes;

  return Array.from(elements || []).map((item: any) => {
    if (item?.dataset?.type === 'mention') {
      const mentionValue = JSON.parse(item.dataset.content);
      return { type: 'mention', content: mentionValue?.data, nodeName: item.nodeName };
    }

    // handles new line by pressing enter
    else if (item.innerText === '\n' && item.nodeName === 'DIV') {
      return { type: 'newline', content: '\n', nodeName: item.nodeName };
    }

    // handles line break by pressing shift + enter
    else if (item.nodeName === 'BR') {
      return { type: 'linebreak', content: '<br>', nodeName: item.nodeName };
    }

    return { type: 'text', content: item.innerText || item.textContent, nodeName: item.nodeName };
  });
};
