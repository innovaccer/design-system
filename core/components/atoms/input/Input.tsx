import * as React from 'react';
import classNames from 'classnames';
import { Tooltip, Icon, Text } from '@/index';
import { IconProps } from '@/index.type';
import { BaseHtmlProps, BaseProps, extractBaseProps } from '@/utils/types';

export type InputType = 'text' | 'password' | 'number' | 'email' | 'tel' | 'url';
export type AutoComplete = 'on' | 'off';
export type Size = 'tiny' | 'regular' | 'large';

export interface InputProps extends BaseProps, BaseHtmlProps<HTMLInputElement> {
  /**
   * Name of the `Input`
   */
  name?: string;
  /**
   * Type of text inside `Input`
   * @default "text"
   */
  type?: InputType;
  /**
   * Value of the `Input` (Used in case of controlled `Input`)
   */
  value?: string;
  /**
   * Adds default value to `Input` (Used in case of uncontrolled `Input`)
   */
  defaultValue?: string;
  /**
   * Text to display when input is empty
   */
  placeholder?: string;
  /**
   * Size of the `Input`
   * @default "regular"
   */
  size?: Size;
  /**
   * Material icon name
   */
  icon?: string;
  /**
   * Label to be displayed inside `Input`
   */
  inlineLabel?: string;
  /**
   * Disables the `Input`, making it unable to type
   */
  disabled?: boolean;
  /**
   * Shows the user that this field id required
   */
  required?: boolean;
  /**
   * Adds autoFocus
   */
  autoFocus?: boolean;
  /**
   * Specifies whether `input field should have autocomplete enabled **(SOON TO BE DEPRECATED)**
   */
  autocomplete?: AutoComplete;
  /**
   * Specifies whether `input field should have autocomplete enabled
   */
  autoComplete?: AutoComplete;
  /**
   * Disables the `Input`, making it unable to type
   */
  readOnly?: boolean;
  /**
   * Valid for number it defines the most negative value in the range of permitted values.
   */
  min?: number;
  /**
   * Valid for text, url, tel, email, and password, it defines the maximum number of characters
   * (as UTF-16 code units) the user can enter into the field.
   */
  max?: number;
  /**
   * Valid for text, url, tel, email, and password, it defines the minimum number of characters
   * (as UTF-16 code units) the user can enter into the entry field.
   */
  minLength?: number;
  /**
   * Valid for text, url, tel, email, and password, it defines the maximum number of characters
   * (as UTF-16 code units) the user can enter into the field.
   */
  maxLength?: number;
  /**
   * The pattern attribute, when specified, is a regular expression that the input's value
   * must match in order for the value to pass constraint validation.
   */
  pattern?: string;
  /**
   * Shows error state in case of failed validation
   */
  error?: boolean;
  /**
   * Text to be rendered in info `Popover`
   */
  info?: string;
  /**
   * minimum width of `Input` component
   *
   * **No default value for `type='number'`**
   * @default 256
   */
  minWidth?: string;
  /**
   * Callback function when user clicks the clear button
   */
  onClear?: (e: React.MouseEvent<HTMLElement>) => void;
  /**
   * Callback function when `Input` text changes
   */
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
  /**
   * Handler to be called when `Input` is clicked
   */
  onClick?: (e: React.MouseEvent<HTMLInputElement>) => void;
  /**
   * Handler to be called when `Input` loses focus
   */
  onBlur?: (e: React.FocusEvent<HTMLInputElement>) => void;
  /**
   * Handler to be called when `Input` gets focus
   */
  onFocus?: (e: React.FocusEvent<HTMLInputElement>) => void;
  /**
   * Custom Icon Component to be passed to Input to replace Clear Icon in the right
   */
  actionIcon?: React.ReactElement<IconProps>;
}

const sizeMapping = {
  tiny: 12,
  regular: 16,
  large: 20,
};

/**
 * ###### Input has two types:
 *  - [Controlled Input](https://reactjs.org/docs/forms.html#controlled-components)
 *  - [Uncontrolled Input](https://reactjs.org/docs/uncontrolled-components.html)
 */
export const Input = React.forwardRef<HTMLInputElement, InputProps>((props, forwardedRef) => {
  const {
    size = 'regular',
    type = 'text',
    minWidth = type !== 'number' ? 256 : undefined,
    readOnly,
    defaultValue,
    name,
    placeholder,
    value,
    icon,
    inlineLabel,
    required,
    error,
    info,
    onChange,
    onClick,
    onClear,
    onBlur,
    onFocus,
    actionIcon,
    className,
    autocomplete,
    autoFocus,
    ...rest
  } = props;

  const ref = React.useRef<HTMLInputElement>(null);

  React.useImperativeHandle(forwardedRef, (): HTMLInputElement => {
    return ref.current as HTMLInputElement;
  });

  React.useEffect(() => {
    if (autoFocus) ref.current?.focus({ preventScroll: true });
  }, []);

  const autoComplete = props.autoComplete || autocomplete;
  const disabled = props.disabled || readOnly;

  const baseProps = extractBaseProps(props);

  const classes = classNames({
    ['Input']: true,
    [`Input--${size}`]: size,
    ['Input--disabled']: disabled,
    ['Input--error']: error
  }, className);

  const inputClass = classNames({
    ['Input-input']: true,
    [`Input-input--${size}`]: size,
  });

  const leftIconClass = classNames({
    ['Input-icon']: true,
    ['Input-icon--left']: true,
    ['Input-icon--disabled']: !value
  });

  const rightIconClass = classNames({
    ['Input-icon']: true,
    ['Input-icon--right']: true
  });

  const trigger = <div className={rightIconClass}><Icon name={'info'} size={sizeMapping[size]} /></div>;

  return (
    <div data-test="DesignSystem-InputWrapper" className={classes} style={{ minWidth }}>
      {inlineLabel && (
        <div className="Input-inlineLabel">
          <Text appearance="subtle">{inlineLabel}</Text>
        </div>
      )}
      {size !== 'tiny' && icon && (
        <div className={leftIconClass}>
          <Icon
            name={icon}
            size={sizeMapping[size]}
          />
        </div>
      )}
      <input
        data-test="DesignSystem-Input"
        {...baseProps}
        {...rest}
        ref={ref}
        name={name}
        type={type}
        defaultValue={defaultValue}
        placeholder={placeholder}
        className={inputClass}
        value={value}
        required={required}
        autoComplete={autoComplete}
        disabled={disabled}
        onChange={onChange}
        onBlur={onBlur}
        onClick={onClick}
        onFocus={onFocus}
      />
      {(!value && !disabled) || (value && disabled) || (defaultValue && disabled)
        ? (
          info && (
            <Tooltip
              position="top"
              tooltip={info}
            >
              {trigger}
            </Tooltip>
          )
        ) : (
          actionIcon
            ? (
              actionIcon
            ) : (
              (onClear && value && !disabled) && (
                <div className={rightIconClass} onClick={e => onClear(e)}>
                  <Icon name={'close'} size={sizeMapping[size]} />
                </div>
              )
            )
        )
      }
    </div>
  );
});

Input.displayName = 'Input';

export default Input;
