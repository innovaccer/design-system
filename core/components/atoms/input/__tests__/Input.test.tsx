import * as React from 'react';
import { render } from '@testing-library/react';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';
import Input, { InputProps as Props } from '../Input';

const iconValue = ['events'];
const inputValue = ['value'];
const StringValue = 'value';
const nameValue = 'name';
const inputType = ['text', 'password', 'number'];
const size = ['tiny', 'regular', 'large'];
const FunctionValue = jest.fn();

describe('Input component', () => {
  const mapper: Record<string, any> = {
    name: valueHelper(nameValue, { required: true }),
    size: valueHelper(size, { required: true, iterate: true }),
    onChangeHandler: valueHelper(FunctionValue, { required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<Input {...attr} />);
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Input component', () => {
  const mapper: Record<string, any> = {
    name: valueHelper(nameValue, { required: true }),
    type: valueHelper(inputType, { required: true, iterate: true }),
    onChangeHandler: valueHelper(FunctionValue, { required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<Input {...attr} />);
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Input component', () => {
  const mapper: Record<string, any> = {
    name: valueHelper(nameValue, { required: true }),
    size: valueHelper(size, { required: true, iterate: true }),
    type: valueHelper('number', { required: true }),
    inlineLabel: valueHelper('USD', { required: true }),
    onChangeHandler: valueHelper(FunctionValue, { required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<Input {...attr} />);
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Input component', () => {
  const mapper: Record<string, any> = {
    name: valueHelper(nameValue, { required: true }),
    value: valueHelper(inputValue, { iterate: true }),
    placeholder: valueHelper('Placeholder', { required: true }),
    icon: valueHelper(iconValue, { iterate: true }),
    onChangeHandler: valueHelper(FunctionValue, { required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<Input {...attr} />);
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Input component', () => {
  const mapper: Record<string, any> = {
    name: valueHelper(nameValue, { required: true }),
    value: valueHelper(StringValue, { required: true }),
    disabled: valueHelper(true, { required: true }),
    icon: valueHelper(iconValue, { iterate: true }),
    onChangeHandler: valueHelper(FunctionValue, { required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<Input {...attr} />);
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Input component', () => {
  const mapper: Record<string, any> = {
    name: valueHelper(nameValue, { required: true }),
    value: valueHelper(StringValue, { required: true }),
    disabled: valueHelper(true, { required: true }),
    icon: valueHelper(iconValue, { iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<Input {...attr} />);
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Input component', () => {
  const mapper: Record<string, any> = {
    name: valueHelper(nameValue, { required: true }),
    value: valueHelper(StringValue, { required: true }),
    error: valueHelper(true, { required: true }),
    icon: valueHelper(iconValue, { iterate: true }),
    onChangeHandler: valueHelper(FunctionValue, { required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<Input {...attr} />);
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});
