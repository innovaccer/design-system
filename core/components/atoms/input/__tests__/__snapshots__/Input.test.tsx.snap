// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component 
  name: "name", placeholder: "Placeholder", onChangeHandler: "[Function]"
 1`] = `
<div
  className="Input Input--regular"
  data-test="DesignSystem-InputWrapper"
  onClick={[Function]}
  style={
    Object {
      "minWidth": 256,
    }
  }
>
  <input
    className="Input-input Input-input--regular"
    data-test="DesignSystem-Input"
    name="name"
    onChangeHandler={[MockFunction]}
    placeholder="Placeholder"
    type="text"
  />
</div>
`;

exports[`Input component 
  name: "name", placeholder: "Placeholder", onChangeHandler: "[Function]", icon: "events"
 1`] = `
<div
  className="Input Input--regular"
  data-test="DesignSystem-InputWrapper"
  onClick={[Function]}
  style={
    Object {
      "minWidth": 256,
    }
  }
>
  <div
    className="Input-icon Input-icon--left Input-icon--disabled"
  >
    <Icon
      name="events"
      size={16}
    />
  </div>
  <input
    className="Input-input Input-input--regular"
    data-test="DesignSystem-Input"
    name="name"
    onChangeHandler={[MockFunction]}
    placeholder="Placeholder"
    type="text"
  />
</div>
`;

exports[`Input component 
  name: "name", placeholder: "Placeholder", onChangeHandler: "[Function]", value: "value"
 1`] = `
<div
  className="Input Input--regular"
  data-test="DesignSystem-InputWrapper"
  onClick={[Function]}
  style={
    Object {
      "minWidth": 256,
    }
  }
>
  <input
    className="Input-input Input-input--regular"
    data-test="DesignSystem-Input"
    name="name"
    onChangeHandler={[MockFunction]}
    placeholder="Placeholder"
    type="text"
    value="value"
  />
</div>
`;

exports[`Input component 
  name: "name", placeholder: "Placeholder", onChangeHandler: "[Function]", value: "value", icon: "events"
 1`] = `
<div
  className="Input Input--regular"
  data-test="DesignSystem-InputWrapper"
  onClick={[Function]}
  style={
    Object {
      "minWidth": 256,
    }
  }
>
  <div
    className="Input-icon Input-icon--left"
  >
    <Icon
      name="events"
      size={16}
    />
  </div>
  <input
    className="Input-input Input-input--regular"
    data-test="DesignSystem-Input"
    name="name"
    onChangeHandler={[MockFunction]}
    placeholder="Placeholder"
    type="text"
    value="value"
  />
</div>
`;

exports[`Input component 
  name: "name", size: "large", onChangeHandler: "[Function]"
 1`] = `
<div
  className="Input Input--large"
  data-test="DesignSystem-InputWrapper"
  onClick={[Function]}
  style={
    Object {
      "minWidth": 256,
    }
  }
>
  <input
    className="Input-input Input-input--large"
    data-test="DesignSystem-Input"
    name="name"
    onChangeHandler={[MockFunction]}
    type="text"
  />
</div>
`;

exports[`Input component 
  name: "name", size: "large", type: "number", inlineLabel: "USD", onChangeHandler: "[Function]"
 1`] = `
<div
  className="Input Input--large"
  data-test="DesignSystem-InputWrapper"
  onClick={[Function]}
  style={
    Object {
      "minWidth": undefined,
    }
  }
>
  <div
    className="Input-inlineLabel"
  >
    <Text
      appearance="subtle"
      size="regular"
    >
      USD
    </Text>
  </div>
  <input
    className="Input-input Input-input--large"
    data-test="DesignSystem-Input"
    name="name"
    onChangeHandler={[MockFunction]}
    type="number"
  />
</div>
`;

exports[`Input component 
  name: "name", size: "regular", onChangeHandler: "[Function]"
 1`] = `
<div
  className="Input Input--regular"
  data-test="DesignSystem-InputWrapper"
  onClick={[Function]}
  style={
    Object {
      "minWidth": 256,
    }
  }
>
  <input
    className="Input-input Input-input--regular"
    data-test="DesignSystem-Input"
    name="name"
    onChangeHandler={[MockFunction]}
    type="text"
  />
</div>
`;

exports[`Input component 
  name: "name", size: "regular", type: "number", inlineLabel: "USD", onChangeHandler: "[Function]"
 1`] = `
<div
  className="Input Input--regular"
  data-test="DesignSystem-InputWrapper"
  onClick={[Function]}
  style={
    Object {
      "minWidth": undefined,
    }
  }
>
  <div
    className="Input-inlineLabel"
  >
    <Text
      appearance="subtle"
      size="regular"
    >
      USD
    </Text>
  </div>
  <input
    className="Input-input Input-input--regular"
    data-test="DesignSystem-Input"
    name="name"
    onChangeHandler={[MockFunction]}
    type="number"
  />
</div>
`;

exports[`Input component 
  name: "name", size: "tiny", onChangeHandler: "[Function]"
 1`] = `
<div
  className="Input Input--tiny"
  data-test="DesignSystem-InputWrapper"
  onClick={[Function]}
  style={
    Object {
      "minWidth": 256,
    }
  }
>
  <input
    className="Input-input Input-input--tiny"
    data-test="DesignSystem-Input"
    name="name"
    onChangeHandler={[MockFunction]}
    type="text"
  />
</div>
`;

exports[`Input component 
  name: "name", size: "tiny", type: "number", inlineLabel: "USD", onChangeHandler: "[Function]"
 1`] = `
<div
  className="Input Input--tiny"
  data-test="DesignSystem-InputWrapper"
  onClick={[Function]}
  style={
    Object {
      "minWidth": undefined,
    }
  }
>
  <div
    className="Input-inlineLabel"
  >
    <Text
      appearance="subtle"
      size="regular"
    >
      USD
    </Text>
  </div>
  <input
    className="Input-input Input-input--tiny"
    data-test="DesignSystem-Input"
    name="name"
    onChangeHandler={[MockFunction]}
    type="number"
  />
</div>
`;

exports[`Input component 
  name: "name", type: "number", onChangeHandler: "[Function]"
 1`] = `
<div
  className="Input Input--regular"
  data-test="DesignSystem-InputWrapper"
  onClick={[Function]}
  style={
    Object {
      "minWidth": undefined,
    }
  }
>
  <input
    className="Input-input Input-input--regular"
    data-test="DesignSystem-Input"
    name="name"
    onChangeHandler={[MockFunction]}
    type="number"
  />
</div>
`;

exports[`Input component 
  name: "name", type: "password", onChangeHandler: "[Function]"
 1`] = `
<div
  className="Input Input--regular"
  data-test="DesignSystem-InputWrapper"
  onClick={[Function]}
  style={
    Object {
      "minWidth": 256,
    }
  }
>
  <input
    className="Input-input Input-input--regular"
    data-test="DesignSystem-Input"
    name="name"
    onChangeHandler={[MockFunction]}
    type="password"
  />
</div>
`;

exports[`Input component 
  name: "name", type: "text", onChangeHandler: "[Function]"
 1`] = `
<div
  className="Input Input--regular"
  data-test="DesignSystem-InputWrapper"
  onClick={[Function]}
  style={
    Object {
      "minWidth": 256,
    }
  }
>
  <input
    className="Input-input Input-input--regular"
    data-test="DesignSystem-Input"
    name="name"
    onChangeHandler={[MockFunction]}
    type="text"
  />
</div>
`;

exports[`Input component 
  name: "name", value: "value", disabled: true
 1`] = `
<div
  className="Input Input--regular Input--disabled"
  data-test="DesignSystem-InputWrapper"
  onClick={[Function]}
  style={
    Object {
      "minWidth": 256,
    }
  }
>
  <input
    className="Input-input Input-input--regular"
    data-test="DesignSystem-Input"
    disabled={true}
    name="name"
    type="text"
    value="value"
  />
</div>
`;

exports[`Input component 
  name: "name", value: "value", disabled: true, icon: "events"
 1`] = `
<div
  className="Input Input--regular Input--disabled"
  data-test="DesignSystem-InputWrapper"
  onClick={[Function]}
  style={
    Object {
      "minWidth": 256,
    }
  }
>
  <div
    className="Input-icon Input-icon--left"
  >
    <Icon
      name="events"
      size={16}
    />
  </div>
  <input
    className="Input-input Input-input--regular"
    data-test="DesignSystem-Input"
    disabled={true}
    name="name"
    type="text"
    value="value"
  />
</div>
`;

exports[`Input component 
  name: "name", value: "value", disabled: true, onChangeHandler: "[Function]"
 1`] = `
<div
  className="Input Input--regular Input--disabled"
  data-test="DesignSystem-InputWrapper"
  onClick={[Function]}
  style={
    Object {
      "minWidth": 256,
    }
  }
>
  <input
    className="Input-input Input-input--regular"
    data-test="DesignSystem-Input"
    disabled={true}
    name="name"
    onChangeHandler={[MockFunction]}
    type="text"
    value="value"
  />
</div>
`;

exports[`Input component 
  name: "name", value: "value", disabled: true, onChangeHandler: "[Function]", icon: "events"
 1`] = `
<div
  className="Input Input--regular Input--disabled"
  data-test="DesignSystem-InputWrapper"
  onClick={[Function]}
  style={
    Object {
      "minWidth": 256,
    }
  }
>
  <div
    className="Input-icon Input-icon--left"
  >
    <Icon
      name="events"
      size={16}
    />
  </div>
  <input
    className="Input-input Input-input--regular"
    data-test="DesignSystem-Input"
    disabled={true}
    name="name"
    onChangeHandler={[MockFunction]}
    type="text"
    value="value"
  />
</div>
`;

exports[`Input component 
  name: "name", value: "value", error: true, onChangeHandler: "[Function]"
 1`] = `
<div
  className="Input Input--regular Input--error"
  data-test="DesignSystem-InputWrapper"
  onClick={[Function]}
  style={
    Object {
      "minWidth": 256,
    }
  }
>
  <input
    className="Input-input Input-input--regular"
    data-test="DesignSystem-Input"
    name="name"
    onChangeHandler={[MockFunction]}
    type="text"
    value="value"
  />
</div>
`;

exports[`Input component 
  name: "name", value: "value", error: true, onChangeHandler: "[Function]", icon: "events"
 1`] = `
<div
  className="Input Input--regular Input--error"
  data-test="DesignSystem-InputWrapper"
  onClick={[Function]}
  style={
    Object {
      "minWidth": 256,
    }
  }
>
  <div
    className="Input-icon Input-icon--left"
  >
    <Icon
      name="events"
      size={16}
    />
  </div>
  <input
    className="Input-input Input-input--regular"
    data-test="DesignSystem-Input"
    name="name"
    onChangeHandler={[MockFunction]}
    type="text"
    value="value"
  />
</div>
`;
